### ISSUE TITLE
Marks of the Valiant/Champion are not awarded if the Valiant/Champion was finished off with Charge

### ISSUE THREAD

**Issue by Nocturne1501** [TrinityCore/TrinityCore#22251]
**Description:**

Ending the fight with a Valiant or Champion at the Argent Tournament via Charge will not earn the player a Mark of the Champion or Mark of the Valiant.

**Current behaviour:**

As above.

**Expected behaviour:**

A Mark of the Valiant/Mark of the Champion should be awarded as normal.

**Steps to reproduce the problem:**

1. .quest add 13790
2. Mount up, fight a Champion
3. Make sure to deal the "killing blow" with Charge

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

TrinityCore rev. 038ad52dce3d+ 2018-08-11 03:16:41 +0000 (3.3.5 branch) (Unix, RelWithDebInfo, Static)

**Operating system:**

Ubuntu 18.04

**Comment by Aokromes** [comment]
update your core to at least 3f7ad3b.

**Comment by Nocturne1501** [comment]
I did git pull origin 3.3.5 literally 30 minutes ago.

**Comment by Killyana** [comment]
Yes looks broken again, it was fixed mounts ago, but on the current version it's not.
